### PR TITLE
Add min_x, max_x ,bin_width to circular_average function in roi module

### PR DESCRIPTION
**Pull request by sameera2004**

### CODE REVIEW

File: skxray/core/roi.py [hunk 1]
```diff
@@ -403,11 +403,10 @@ def mean_intensity(images, labeled_array, index=None):
     return mean_intensity, index
 
 
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=None):
+def circular_average(image, calibrated_center, threshold=0, min_x=None,
+                     max_x=None, bin_width=1., pixel_size=(1, 1), nx=None):
```
Comment (tacaswell) [inline skxray/core/roi.py]: Please don't change the kwarg order, that is a back-compatibility break.

File: skxray/core/roi.py [hunk 2]
```diff
@@ -403,11 +403,10 @@ def mean_intensity(images, labeled_array, index=None):
     return mean_intensity, index
 
 
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=None):
+def circular_average(image, calibrated_center, threshold=0, nx=None,
+                     pixel_size=(1, 1),  min_x=None, max_x=None, bin_width=1.):
```
Comment (tacaswell) [inline skxray/core/roi.py]: :-1: on bin width.  It leads to ambiguity if min_x, max_x and bin_width are all specified.

**Comment by sameera2004** [comment]
@tacaswell travis is working changed the lmfit version to 0.8.3. Ready for review

File: skxray/core/roi.py [hunk 3]
```diff
@@ -430,11 +432,23 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
...
+
+    if min_x is None:
+        min_x = np.min(radial_val)
+    else:
+        min_x = min_x * pixel_size[0]
+    if max_x is None:
+        max_x = np.max(radial_val)
+    else:
+        max_x = max_x * pixel_size[0]
+    if nx is None:
```
Comment (tacaswell) [inline skxray/core/roi.py]: This is changing the default behavior

Comment (sameera2004) [inline skxray/core/roi.py]: @tacaswell I changed nx back to default value

File: skxray/core/roi.py [hunk 4]
```diff
@@ -430,11 +432,23 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape,
-                                   pixel_size)
+    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
+
+    if min_x is None:
```
Comment (tacaswell) [inline skxray/core/roi.py]: I don't think you need this logic, it is duplicated in https://github.com/tacaswell/scikit-xray/blob/master/skxray/core/utils.py#L588

File: skxray/core/roi.py (hunk 3 above)
Comment (tacaswell) [inline skxray/core/roi.py]: This check should probably be deleted or the behavior of passing in `None` causes 1 pixel bins to be documented.

**Comment by sameera2004** [comment]
@tacaswell documented the API changes ready for review

**Comment by sameera2004** [comment]
@tacaswell thanks!

**Pull request merged**